cmd/validatepcap/parse.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os/exec"
+	"strconv"
+)
+
+type ClientHello struct {
+	version           uint16
+	supportsDC        bool
+	serverName        string
+	supportedVersions []uint16
+}
+
+type ServerHello struct {
+	version uint16
+}
+
+type Transcript struct {
+	clientHello ClientHello
+	serverHello ServerHello
+}
+
+func parsePcap(tsharkPath string, pcapPath string, keylogPath string) Transcript {
+	rawJSON, err := exec.Command(tsharkPath,
+		"-r", pcapPath,
+		"-d", "tcp.port==4433,ssl",
```

Comment:
This should not be needed, TLS is explicitly recognized through a heuristic if there is no other port override. It does not hurt though. I would however suggest changing the deprecated `ssl` to `tls`. The arguments can also be joined, whether you want to do it is a style question:

```suggestion
		"-dtcp.port==4433,tls",
```